src/Sentry/Sentry.csproj

```diff
@@ -5,6 +5,8 @@
     <NoWarn Condition="'$(TargetFramework)' == 'netstandard2.0'">$(NoWarn);RS0017</NoWarn>
     <CLSCompliant Condition="'$(TargetPlatformIdentifier)' == ''">true</CLSCompliant>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IsTrimmable>true</IsTrimmable>
+    <DefineConstants>$(DefineConstants);TRIMMABLE</DefineConstants>
```

Comment:
I does yeah... I didn't intend for this to be merged into the feat/4.0.0 branch - it was just a quick/easy hack to enable trimming so that I could get a sense for the challenges and start to try to work through those. 

I was scratching my head to think how we could do this for realz and haven't come up with any good options yet. We might, as you suggested, have to have [separate packages for AOT](https://learn.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks#architecture-specific-folders) and Managed. 

I'm not sure how that is done in MSBuild though. When targeting multiple frameworks, there's a `TargetFrameworks` property... but I don't know of the equivalent for `PlatformTarget`.

We can always pass this in via an environment variable or the command line, which will work well for our CI workflows on GitHub, but that seems like a clumsy experience for us on our development machines... particularly for testing (where again, [multiple frameworks are well supported](https://xunit.net/docs/getting-started/netcore/cmdline#multi-targeting) but multiple platforms conspicuously not mentioned 🤔).